### ISSUE TITLE
Template string with object with template string inside

### ISSUE THREAD

**Issue by scastiel** [babel/babel-eslint#537]
The parsing fails when parsing a piece of code containing:
* a template string
* in this template string: an object
* in this object: another template string

Example: 
```js
`${ { a:`${2}` } }`
```

The inner template string is not converted correctly when passing into `convertTemplateType`.

I forked the repo and added a new test that currently fails to reproduce the problem:  https://github.com/scastiel/babel-eslint/blob/master/test/babel-eslint.js#L157-L159
